posts/bayesian-model-comparison/index.qmd

```diff
@@ -0,0 +1,470 @@
+In the real world, it is usually the case that we want to reason about situations and make decisions based on costs and errors. 
+Phrased differently, the choice of the cost measure should depend on how the system is going to be used, rather than on any inherent specification of the training process.
+The issue with doing this is that it is hard. 
+It is hard because specifying a cost-aware loss function is non-trivial, because cost-specifications are domain specific, and becuase even in the case of roughly knowing what the costs are, using this information is hard, i.e. the specified weigthed cost may be a difficult objective for optimisers to work with.
+However if we can specify costs then decision making based on these would be the best way to work as this allows one to take into consideration the utility of the decision that will be made.
+
+In our situation, in order to do cost-sensitive analysis/decision making, all we need is to specify a cost function -- we won't need to run an optimisation on this objective function. 
+This is a function that defines the loss (or cost) we incur in making a given decision (e.g. the wrong decision).
+A typical example is whether to give more importance to a false positive or a false negative.
+For our given situation and for the sake of exposition let's assume that the random forest model takes 4 times longer to train. 
+There are three decisions we can make 
+
+1) SVM is better than RF
+2) SVM is equivalent to RF
+3) SVM is worse than RF
+
+Consider the following cost-matrix:
+
++---------------+---------------+--------------------+--------------------+
+|               | SVM is better | are equivalent     | RF is better       |
++===============+===============+====================+====================+
+| Choose SVM    | 0             | -5                 | 2                  |
+|               |               |                    |                    |
++---------------+---------------+--------------------+--------------------+
+| Choose RF     | 7             | 5                  | 0                  |
+|               |               |                    |                    |
++---------------+---------------+--------------------+--------------------+
+
+where the $(i, j)$th entry is the cost incurred in making decision $i$ when $j$ is correct.
+Here we have a $2 \times 3$ matrix as we only consider the options of selecting the SVM or the RF, no abstention or anything else.
+
+In this case we have:
+
+* cost of choosing SVM is:
+  + 0 if SVM is better, 
+  + -5 if they are equivalent as we save compute time, 
+  + 2 if RF is better as we would lose performance
+
+* cost of choosing RF is:
+  + 7 if SVM is better because we pay for computational and performance cost, 
+  + 5 if they are equivalent as we add to the compute time, 
+  + 0 if RF is indeed better
+
+
+The **expected cost** can then be obtained by multiplying the cost_matrix with the posterior probabilities
```

Comment:
i find it odd to think of/refer to the cumulative probabilities as posterior, given we dont set priors and we dont compute them as part of the modelling

Reply:
The t-dist here is a posterior though, so we are getting the probabilities given by volumes in the posterior. Am I misunderstanding?

Reply:
what is a posterior? 

i so wanted to just comment w that! i just mean that for this distribution (over {M1<M2, M1=M2, M1>M2}) we have no prior. it feels like a "derived" distribution from the (main) posterior. but i get your point, and happy to leave!

Reply:
Aha, no yes, you are right. We derive this, that's true. I guess you could still call this P(M1 > M2 |D) and not talk about how you would get this directly. After all we could do the same by getting P(M1>M2) based on the Normal prior. Then we perform inference, get a t-dist, then derive the probability.

Reply:
Do you want me to make this clearer?

Reply:
isn't the probability actually P(X|D)  where X \in {M1<M2, M1=M2, M1>M2}, or are we really thinking of as 3 probabilities. i guess above we are.

i think if we have a quick fix could be good. but it's minor. how about 

> The **expected cost** can then be obtained by multiplying the cost_matrix with the probabilities P(M1 > M2 |D), P(M1 ~ M2 |D), and P(M1 < M2 |D)

Reply:
I'll clarify 👍 . I stress though that these are indeed posteriors. Further, they can be considered as cumulative of the actual posterior.  Once you have those then you can reformulate these are parameters of a multinoulli distribution, just like you can take the cumulative of a gaussian and feed it as parameter of a Bernoulli distribution.